index.js

```diff
@@ -199,6 +200,67 @@ client.on("interactionCreate", async (interaction) => {
             await interaction.reply("Failed to pin the message. Please check the message ID and try again.");
         }
     }
+    // Message count:
+    if (commandName === "messagecount") {
+        const user = options.getUser("user");
+        if (!user) {
+            await interaction.reply("User not found.");
+            return;
+        }
+
+        const { id: userId} = user;
+        const { user: { id: targetId } } = interaction;
+        
+        if (userId === client.user.id || userId === targetId) {
+            await interaction.reply("You cannot count messages to yourself or the bot.");
+            return;
+        }
+
+        await interaction.deferReply();
+
+        const channel = interaction.channel;
+        let messageCount = 0;
+        let lastMessageId = null;
+        let reachedTargetUser = false;
+
+        console.log(`Counting messages from ${interaction.user.username} to ${user.username}...`);
+
+        while (true) {
```

Comment:
Instead of using a boolean flag and a counter variable to keep track of this, you could refactor this whole `while` loop into its own function and then simply return the count instead. This would be make the implementation quite a bit cleaner and help break up this huuuge function

Reply:
Will refactor in followup PR